### PR TITLE
Locale negotiation and selection

### PR DESCRIPTION
**Pull request by dve**
I added to small classes to negotiate the "correct" locale by 
  * a user selection
  * the browser accept header
  * the available locales

There is also a select component for the supported locales in their native language.

### CODE REVIEW

File: src/main/java/org/vaadin/viritin/components/LocaleSelect.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@SuppressWarnings("serial")
+public class LocaleSelect extends TypedSelect<Locale> {
+    public LocaleSelect() {
+        setCaptionGenerator(new CaptionGenerator<Locale>() {
+            @Override
+            public String getCaption(Locale option) {
+                return option.getDisplayName(option);
+            }
+        });
+        setImmediate(true);
```
Comment (mstahv) [inline src/main/java/org/vaadin/viritin/components/LocaleSelect.java]: setImmediate is most often not needed with Vaadin anymore since 7.3

**Comment by mstahv** [comment]
This looks really handy, Thanks! Lets try to get some input from others developers as well who have done similar stuff. I could try to ask couple of our consultants, if you have friends using Vaadin, try to ask them as well. Maybe you could advertise this change on Vaadin forum or Google + channel as well?

**Comment by dve** [comment]
Added a comment to the forum: https://vaadin.com/forum#!/thread/9784602

**Comment by mstahv** [comment]
I made some changes to make the build pass with Java 7, but I think I also broke something as the tests didn't pass. Was in a bit of a hurry as I wanted cut a release.  Would you review my changes and post the necessary fixes?

**Comment by dve** [comment]
I merged some fixes I've made in my repo and prepare a pull request

**Pull request merged**